config/job.js

```diff
@@ -4,6 +4,42 @@ const Annotations = require('./annotations');
 const Joi = require('joi');
 const Regex = require('./regex');
 
+// ref. https://github.com/hapijs/joi/blob/v13.3.0/API.md#extendextension
+const sdJoi = Joi.extend(joi => ({
+    base: joi.string(),
+    name: 'string',
+    language: {
+        commitBranch: 'needs to be a commit branch string/regex'
+    },
+    rules: [
+        {
+            name: 'commitBranch',
+            validate(params, value, state, options) {
+                const regexPos = 3;
+                const matched = Regex.TRIGGER.exec(value);
+                const brFilter = matched[regexPos];
```

Comment:
I think we should also add comments to these parsing.

Reply:
Sorry my variable name is not good I will fix the name regexPos to specificBranchPos.
Then, it give you sense which part this code try to get. 
After this correction and see the Regex.TRIGGER, maybe you can understand easily.

What do you think?

Reply:
I added comments.